dvc/remote/http.py

```diff
@@ -43,24 +27,27 @@ def __init__(self, repo, config):
         self.path_info = self.path_cls(url) if url else None
 
     def _download(self, from_info, to_file, name=None, no_progress_bar=False):
-        callback = None
-        if not no_progress_bar:
-            total = self._content_length(from_info.url)
-            if total:
-                callback = ProgressBarCallback(name, total)
-
         request = self._request("GET", from_info.url, stream=True)
-
-        with open(to_file, "wb") as fd:
-            transferred_bytes = 0
-
-            for chunk in request.iter_content(chunk_size=self.CHUNK_SIZE):
-                fd.write(chunk)
-                fd.flush()
-                transferred_bytes += len(chunk)
-
-                if callback:
-                    callback(transferred_bytes)
+        total = self._content_length(from_info.url)
+        if name is None:
+            name = from_info.url
+        leave = False
+        # TODO: persistent progress only for "large" files?
```

Comment:
Could you please elaborate?

Reply:
Also, we try not to leave TODOs in the code itself, instead we create an issue.

Reply:
I meant maybe we should clear away/overwrite progress bars (`leave=False`) only if the files are small. But perhaps `leave=True` for files greater than `LARGE_FILE_SIZE`?

Reply:
Ah, so right now user will simply see N_WORKERS progress bars no matter how many files are downloaded, because tqdm will remove the old ones, right? In that case it seems great to leave it that way, since our old approach was just flooding the terminal :slightly_smiling_face:

Reply:
Yes, just up to `N_WORKERS` bars, which should disappear as they complete. However I've just updated to leave them behind if they correspond to files larger than `LARGE_FILE_SIZE`. Seems like a reasonable thing to do. What else would be the point of the `LARGE_FILE_SIZE` variable? Be more verbose, surely, as in its current usage when calculating md5sums.

Reply:
actually it may not have cleared away all bars - just pushed another update. Also https://github.com/tqdm/tqdm/issues/509 could in theory be a problem though it wouldn't affect `dvc` unless it was used in a python script that used other classes of `tqdm`

Reply:
LARGE_FILE_SIZE is used in checksum progress bars to keep the user updated on what is going on, it is not really used in push/pull/etc, and I think we could safely not leave progress bars for large files, since they've already done their job keeping the user updated while the download/upload is going on.

Reply:
removed